### DISCUSSION TITLE
Ordering  or filtering by metadata date field

### DISCUSSION THREAD

**Discussion by spppaul** [neon-sunset/Pinecone.NET#113] [General]
Is there any way to filter or order by a metadata date field? I'm trying to give more weight to more recent items that are stored, is anything like that possible?

**Comment by neon-sunset** [comment]
Depends on *where* you expect the ordering to occur.

Pinecone only has metadata filtering API, but if you query all relevant vectors, then you could simply `.OrderBy` `ScoredVector[]` returned by `index.Query` as any other collection.

**Comment by spppaul** [comment]
I'm looking specifically for server side, since there is no date type, how would filtering based on the date string in the metadata work, are there any examples for that?

**Reply by neon-sunset** [reply]
Pinecone.NET follows standard metadata API (I suppose with some differences as far as syntax goes): https://docs.pinecone.io/guides/data/filter-with-metadata

Examples: https://github.com/neon-sunset/Pinecone.NET/discussions/31#discussioncomment-6302396

**Reply by neon-sunset** [reply]
@spppaul does this address your question?

**Comment by spppaul** [comment]
Yes, I'll filter on date string  of yyyyMMdd or component numbers, thanks
